server/service/endpoint_setup.go

```diff
@@ -85,3 +118,229 @@ func makeSetupEndpoint(svc fleet.Service, logger kitlog.Logger) endpoint.Endpoin
 		}, nil
 	}
 }
+
+// applyStarterLibrary downloads the starter library from GitHub
+// and applies it to the Fleet server using an authenticated client.
+func applyStarterLibrary(
+	ctx context.Context,
+	serverURL string,
+	token string,
+	logger kitlog.Logger,
+	httpClientFactory func(opts ...fleethttp.ClientOpt) *http.Client,
+	clientFactory func(serverURL string, insecureSkipVerify bool, rootCA, urlPrefix string, options ...ClientOption) (*Client, error),
+	// For testing only - if provided, this function will be used instead of client.ApplyGroup
+	mockApplyGroup func(ctx context.Context, specs *spec.Group) error,
```

Comment:
Is it reasonable inject a test dependency like this? I figured better than not testing the function at all.